geoportailv3/static/js/share/shorturl.js

```diff
@@ -0,0 +1,105 @@
+
+/**
+ * @fileoverview This file provides a shorturl directive
+ * This directive is used to create a short url panel in the page.
+ *
+ * Example:
+ *
+ * <app-shorturl app-shorturl-visible="::mainCtrl.visible"></app-shorturl>
+ *
+ */
+goog.provide('app.shorturlDirective');
+
+goog.require('app');
+goog.require('ngeo.Location');
+
+
+/**
+ * @param {string} appShorturlTemplateUrl Url to share template
+ * @return {angular.Directive} The Directive Definition Object.
+ * @ngInject
+ */
+app.shorturlDirective = function(appShorturlTemplateUrl) {
+  return {
+    restrict: 'E',
+    scope: {
+      'visible': '=appShorturlVisible'
+    },
+    controller: 'AppShorturlController',
+    controllerAs: 'ctrl',
+    bindToController: true,
+    templateUrl: appShorturlTemplateUrl
+  };
+};
+
+app.module.directive('appShorturl', app.shorturlDirective);
+
+
+
+/**
+ * @ngInject
+ * @constructor
+ * @param {ngeo.Location} ngeoLocation
+ * @param {angular.$http} $http
+ * @param {angular.Scope} $rootScope
+ * @param {string} shorturlServiceUrl
+ * @export
+ */
+app.ShorturlDirectiveController =
+    function(ngeoLocation, $http, $rootScope, shorturlServiceUrl) {
+  /**
+   * @type {ngeo.Location}
+   * @private
+   */
+  this.ngeoLocation_ = ngeoLocation;
+
+  /**
+   * @type {string}
+   * @export
```

Comment:
For the moment we don't use `@export` for properties that are not methods on the prototype. So I'd remove this and use `this['url']` instead.